### PR TITLE
Fix/7.6.0 constructor injection upgrade bug

### PR DESCRIPTION
**Pull request by joshheald**
Related to #7464 

#### Changes proposed in this Pull Request

There is an issue which breaks the upgrade process, causing a fatal error on upgrade, when REST API constructors are changed.

This reverts a constructor change and gets the token_service via a static function as a workaround.

#### Testing instructions

* Using a JN site with WooPayments installed
* Upgrade the site to use this version of WooPayments (`npm run build` this branch, and upload using the plugin installer in WP-admin
* Verify that there is no fatal error during the upgrade

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8794` or branch name `fix/7.6.0-constructor-injection-upgrade-bug` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7.6.0-constructor-injection-upgrade-bug) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: d14ea52e7ff4ae53fd4967fbb9c3ed459e4a66ff
- Build time: 2024-05-07 18:26:57 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/admin/class-wc-rest-payments-orders-controller.php [hunk 1]
```diff
@@ -237,7 +235,7 @@ function_exists( 'wcs_get_subscriptions_for_order' ) &&
 										function_exists( 'wcs_is_manual_renewal_required' ) &&
 										wcs_order_contains_subscription( $order_id );
 				if ( $has_subscriptions ) {
-					$token = $this->token_service->add_payment_method_to_user( $generated_card, $order->get_user() );
+					$token = WC_Payments::get_token_service()->add_payment_method_to_user( $generated_card, $order->get_user() );
```
Comment (gpressutto5) [inline includes/admin/class-wc-rest-payments-orders-controller.php:231]: `private $token_service` is still declared in this class. Please, remove it.

File: includes/class-wc-payments.php [hunk 2]
```diff
@@ -1325,6 +1325,26 @@ public static function get_order_service(): WC_Payments_Order_Service {
 		return self::$order_service;
 	}
 
+	/**
+	 * Returns the token service instance.
+	 *
+	 * @return WC_Payments_Token_Service
+	 */
+	public static function get_token_service(): WC_Payments_Token_Service {
+		return self::$token_service;
```
Comment (gpressutto5) [inline includes/class-wc-payments.php:1334]: Declare this private static property.

**Review by gpressutto5 — CHANGES_REQUESTED** [review]

File: includes/class-wc-payments.php (hunk 2 above)
Comment (joshheald) [reply includes/class-wc-payments.php:1334]: @gpressutto5 Sorry, I'm not sure what you mean here...

`token_service` is already a private static property on this class (line 97)... and if you mean making this function private, I'm confused. Surely that would prevent the rest-payments-orders-controller from using it, which is why we added it in the first place? The other similar getters are all public functions too.

Excuse my ignorance if I'm missing something obvious... I'm here to learn 😄

Comment (gpressutto5) [reply includes/class-wc-payments.php:1334]: You're right, sorry I missed that!

**Review by gpressutto5 — APPROVED** [review]

**Pull request merged**